### PR TITLE
ERC20 test coverage

### LINKED ISSUE #105: Complete ERC20 coverage tests
### Problem

The repo currently has incomplete coverage on ERC20 cases that confirms methods can be utilized

### Solution

Complete test cases coverage

Update [ERC20Contract.js](https://github.com/hashgraph/hedera-smart-contracts/blob/main/test/hts-precompile/erc-20/ERC20Contract.js) with simple tests using ethers to verify 
- [ ] [balanceOf](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/examples/erc-20/ERC20Contract.sol#L25)
- [ ] [transfer](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/examples/erc-20/ERC20Contract.sol#L29)
- [ ] [allowance](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/examples/erc-20/ERC20Contract.sol#L33)
- [ ] [approve](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/examples/erc-20/ERC20Contract.sol#L37)
- [ ] [delegateTransfer](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/examples/erc-20/ERC20Contract.sol#L41)
- [ ] [transferFrom](https://github.com/hashgraph/hedera-smart-contracts/blob/main/contracts/hts-precompile/examples/erc-20/ERC20Contract.sol#L45)

Note: It is not necessary to cover every edge case like we do on the [relay](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/server/tests/acceptance/erc20.spec.ts). This should be simple cases to verify calls can be made .
In a sense it also showcases how DApps might use ethers to make these calls 

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by Ivo-Yankov**
Signed-off-by: Ivo Yankov <ivo@devlabs.bg>

**Description**:
- Adds the methods `delegateApprove` and `delegateTransferFrom` to `ERC20Contract.sol`
- Adds tests for all contract methods

**Related issue(s)**:

Fixes #105 

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: test/hts-precompile/erc-20/ERC20Contract.js [hunk 1]
```diff
@@ -30,6 +40,139 @@ describe("ERC20Contract tests", function () {
...
+
+    expect(contractOwnerBalance).to.exist;
+    expect(contractOwnerBalance.toNumber()).to.eq(TOTAL_SUPPLY - INITIAL_BALANCE);
+    expect(wallet1Balance).to.exist;
+    expect(wallet1Balance.toNumber()).to.eq(INITIAL_BALANCE);
+    expect(wallet2Balance).to.exist;
+    expect(wallet2Balance.toNumber()).to.eq(0);
+  });
+
+  it("should NOT be able to use transfer", async function () {
```
Comment (Nana-EC) [inline test/hts-precompile/erc-20/ERC20Contract.js:59]: Q: Shouldn't there also be a successful example of this?

File: test/hts-precompile/erc-20/ERC20Contract.js [hunk 2]
```diff
@@ -30,6 +40,139 @@ describe("ERC20Contract tests", function () {
...
+    await tx.wait();
+
+    const wallet1BalanceAfter = await erc20Contract.balanceOf(tokenAddress, signers[0].address);
+    const wallet2BalanceAfter = await erc20Contract.balanceOf(tokenAddress, signers[1].address);
+
+    expect(wallet1BalanceBefore.toNumber() - amount).to.eq(wallet1BalanceAfter.toNumber());
+    expect(wallet2BalanceBefore.toNumber() + amount).to.eq(wallet2BalanceAfter.toNumber());
+  });
+
+  it("should NOT be able to use approve", async function () {
```
Comment (Nana-EC) [inline test/hts-precompile/erc-20/ERC20Contract.js:102]: Q: Shouldn't there also be a successful example of this?

File: test/hts-precompile/erc-20/ERC20Contract.js [hunk 3]
```diff
@@ -30,6 +40,139 @@ describe("ERC20Contract tests", function () {
...
+    expect(allowanceBefore.toNumber()).to.eq(0);
+
+    const tx = await erc20Contract.connect(signers[0]).delegateApprove(tokenAddress, signers[1].address, approvedAmount, {gasLimit: 1_000_000});
+    await tx.wait();
+
+    const allowanceAfter = await erc20Contract.allowance(tokenAddress, signers[0].address, signers[1].address);
+    expect(allowanceAfter.toNumber()).to.eq(approvedAmount);
+  });
+
+  it("should NOT be able to use transferFrom", async function () {
```
Comment (Nana-EC) [inline test/hts-precompile/erc-20/ERC20Contract.js:137]: Q: Shouldn't there also be a successful example of this?

**Review by Nana-EC — CHANGES_REQUESTED** [review]
LG, curious about the positive cases

File: test/hts-precompile/erc-20/ERC20Contract.js (hunk 1 above)
Comment (Ivo-Yankov) [reply test/hts-precompile/erc-20/ERC20Contract.js:59]: Those HTS methods cannot be called directly from a contract, that is why we have added the `delegate` methods, although they are not part of the ERC20 standard. The same is true for this [PR](https://github.com/hashgraph/hedera-smart-contracts/pull/124)

**Review by Nana-EC — APPROVED** [review]
